src/sql/workbench/contrib/notebook/browser/find/notebookFindModel.ts

```diff
@@ -535,7 +535,7 @@ export class NotebookFindModel extends Disposable implements INotebookFindModel
 
 	private searchFn(cell: ICellModel, exp: string, matchCase: boolean = false, wholeWord: boolean = false, maxMatches?: number): NotebookRange[] {
 		let findResults: NotebookRange[] = [];
-		if (cell.cellType === 'markdown' && cell.isEditMode && typeof cell.source !== 'string') {
+		if (cell.cellType === 'markdown' && (cell.showMarkdown || cell.currentMode === CellEditModes.SPLIT) && typeof cell.source !== 'string') {
```

Comment:
Would suggest adding a comment here explaining that this is for searching the markdown portion of the cell.

Although in general I think this logic could use some cleaning up to make it clearer what the different types of content you're searching could be : 

1. markdown cell & markdown pane visible -> search cell.source
2. markdown cell & WYSIWYG pane visible -> search cell.renderedOutputTextContent
2. non-markdown cell -> search source content directly

The first two and the last one are exclusive (cellType can only be one or the other)

So the logic would look something like 

```
if(cellType === 'markdown') {
   if(<markdown pane visible>) { 
        // search cell.source
   } 
   if(<rich text pane visible>) {
       // search cell.rederedOutputTextContent
   }
} else { 
    // search cell.source
}
```

This would be a lot easier to follow and understand. (you can make helper functions to combine the search logic for stuff like searching cell.source since those would likely be the same).

Reply:
Yep, I'll do that refactor in my next PR of Find (highlight all terms).